web/client/plugins/map/mapsettings/MapSettings.jsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2022, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect, useState } from 'react';
+import { connect } from 'react-redux';
+import { FormGroup, Checkbox, ControlLabel } from 'react-bootstrap';
+import Message from '../../../components/I18N/Message';
+import { createSelector } from 'reselect';
+import { updateSceneConfig } from '../../../actions/map';
+import ConfigUtils from '../../../utils/ConfigUtils';
+
+import { mapSelector } from '../../../selectors/map';
+import { mapTypeSelector, isCesium as isCesiumSelector } from '../../../selectors/maptype';
+
+const mapStateToProps = createSelector(
+    mapSelector,
+    mapTypeSelector,
+    isCesiumSelector,
+    (map, mapType, isCesium) => ( { map, mapType, isCesium })
+);
+
+const actions = {
+    updateConfigAction: updateSceneConfig
+};
+
+const Component = ({
+    map,
+    mapType,
+    isCesium,
+    updateConfigAction
+}) => {
+    const [mapOptions, setMapOptions] = useState({});
+
+    useEffect(() => {
+        if (map) {
+            setMapOptions({
+                ...ConfigUtils.getConfigProp("defaultMapOptions")[mapType],
+                ...map.mapOptions
```

Comment:
Here is missing the Map.cfg.mapOptions default configuration If compared with options merge inside the Map plugin. We could apply this change inside the Settings plugin

```diff
        return Object.keys(settingsFirst)
            .filter(this.isEnabled)
            .map((setting) => settingsFirst[setting])
-           // TODO: here every item (item.tool) we emit should have a "key" property
-           .concat(this.props.items.map((item) => item.tool))
+           .concat(this.props.items.map((item) =>
+               castArray(item.tool)
+                   .map((tool, idx) => cloneElement(tool, { ...item.cfg, key: `${item.name}-${idx}` })))
+           )
            .concat(
                Object.keys(settingsLast)
                    .filter(this.isEnabled)
                    .map((setting) => settingsLast[setting])
            );
```

- cloneElement from react
- castArray from lodash

with item.cfg spreat to the tool props we will get mapOptions from cfg inside this component so we can merge it with the other map options. Note the mapOptions from cfg is grouped with mapType